main.cpp

```diff
@@ -549,5 +612,61 @@ int main()
 
     divider();
 
+    std::cout << "The apply() function applies the current value of the current object to itself.\n\n";
+
+    FloatType ftA(4.5f);
+
+    std::cout << "FtA is currently: " << static_cast<float>(ftA) << std::endl;
+    
+    ftA.apply( [&]( float &a ) -> FloatType&
+    {
+        a += a;
+        return ftA;    
+    } );
+    
+    std::cout << "FtA applied to itself by lambda is: " << static_cast<float>(ftA) << std::endl;
+
+    ftA.apply( updateValue );
+
+    std::cout << "FtA applied to itself via function pointer is: " << static_cast<float>(ftA) << std::endl;
+
+    divider();
+
+    DoubleType dtA(7.894561);
+
+    std::cout << "dtA is currently: " << static_cast<double>(dtA) << std::endl;
+    
+    dtA.apply( [&]( double &a ) -> DoubleType&
+    {
+        a += a;
+        return dtA;    
+    } );
+    
+    std::cout << "dtA applied to itself by lambda is: " << static_cast<double>(dtA) << std::endl;
+
+    dtA.apply( updateValue );
+
+    std::cout << "dtA applied to itself via function pointer is: " << static_cast<double>(dtA) << std::endl;
+
+    divider();
+
+    IntType itA(495);
+
+    std::cout << "itA is currently: " << static_cast<int>(itA) << std::endl;
+    
+    itA.apply( [&]( int &a ) -> IntType&
```

Comment:
- [ ] Don't capture more than you need to!  You are capturing **everything** in `main()` by reference.